runtime/lua/vim/lsp/codelens.lua

```diff
@@ -214,14 +215,19 @@ end
 --- |lsp-handler| for the method `textDocument/codeLens`
 ---
 function M.on_codelens(err, result, ctx, _)
-  assert(not err, vim.inspect(err))
+  if err then
+    active_refreshes[ctx.bufnr] = nil
+    local _ = log.error() and log.error("codelens", ctx.client_id, err)
+    return
+  end
```

Comment:
I figured logging the error and silently returning would be more in-line with what I'd personally expect. I also think it improves the overall experience, for example `rust_analyzer` returns errors when trying to retrieve codelenses before its cargo metadata check has resolved. With the error approach, this is pretty disruptive as it more or less hijacks the entire editor